crates/ruma-html/src/sanitizer_config/clean.rs

```diff
@@ -475,3 +383,173 @@ enum AttributeAction {
     /// Remove the attribute.
     Remove(Attribute),
 }
+
+/// Whether the given HTML element has a list of allowed schemes according to the Matrix
+/// specification or the compatibility list.
+pub(super) fn has_element_allowed_schemes(element_name: &str) -> bool {
+    // Keep in sync with `spec::allowed_schemes` and `compat::allowed_schemes`.
+    matches!(element_name, "a" | "img")
+}
+
+/// Rules defined in the Matrix specification.
+pub(crate) mod spec {
+    /// Max depth of nested HTML elements allowed by the Matrix specification.
+    pub(super) const MAX_DEPTH: u32 = 100;
+
+    /// The HTML element name for a rich reply fallback.
+    pub(super) const RICH_REPLY_ELEMENT_NAME: &str = "mx-reply";
+
+    /// Whether the given HTML element is allowed in the Matrix specification.
+    pub(super) fn is_element_allowed(element_name: &str) -> bool {
+        matches!(
+            element_name,
+            "del"
+                | "h1"
+                | "h2"
+                | "h3"
+                | "h4"
+                | "h5"
+                | "h6"
+                | "blockquote"
+                | "p"
+                | "a"
+                | "ul"
+                | "ol"
+                | "sup"
+                | "sub"
+                | "li"
+                | "b"
+                | "i"
+                | "u"
+                | "strong"
+                | "em"
+                | "s"
+                | "code"
+                | "hr"
+                | "br"
+                | "div"
+                | "table"
+                | "thead"
+                | "tbody"
+                | "tr"
+                | "th"
+                | "td"
+                | "caption"
+                | "pre"
+                | "span"
+                | "img"
+                | "details"
+                | "summary"
+                | "mx-reply"
+        )
+    }
+
+    /// The replacement element for the given HTML element, if it was previously allowed in the
+    /// Matrix specification.
+    pub(super) fn deprecated_element_replacement(element_name: &str) -> Option<&'static str> {
+        let replacement = match element_name {
+            "font" => "span",
+            "strike" => "s",
+            _ => return None,
+        };
+
+        Some(replacement)
+    }
+
+    /// Whether the given attribute is in the list of allowed attributes for the given HTML element
+    /// according to the Matrix specification.
+    pub(super) fn is_attribute_allowed(element_name: &str, attribute_name: &str) -> bool {
+        match element_name {
+            "span" => match attribute_name {
+                "data-mx-bg-color" | "data-mx-color" | "data-mx-spoiler" | "data-mx-maths" => true,
+                #[cfg(feature = "unstable-msc4286")]
+                "data-msc4286-external-payment-details" => true,
+                _ => false,
+            },
```

Comment:
Sadly e can't use `matches!` for this because it doesn't support `cfg` attributes either.